controllers/tests/createProductController.test.js

```diff
@@ -0,0 +1,392 @@
+   *
+   * Equivalence classes tested:
+   * 1. name: non-empty string (≤100 characters) (valid)
+   * 2. description: non-empty string (≤500 characters) (valid)
+   * 3. price:  valid numeric string (e.g., "10.99", "5") (valid)
+   * 4. category: valid and existent category ID (valid)
+   * 5. shipping: “0” or “1” (valid)
+   * 6. photo: photo where file size <= 1MB
+   */
+  it("Should create products successfully", async () => {
+    await createProductController(req, res);
+
+    // Assertions
+    expect(res.status).toBeCalledWith(201);
+    expect(res.send.mock.lastCall[0].success).toBe(true);
+    expect(res.send.mock.lastCall[0].message).toBe(
+      "Product Created Successfully"
+    );
+  });
+
+  /**
+   * Equivalence classes of field "name" identified:
+   * a. String <= 100 characters and product name does not exist (valid): tested above
+   * b. String <= 100 characters product name already exists (invalid)
+   * c. Empty/null/missing (invalid)
+   * d. String > 100 characters (invalid)
+   *
+   * Classes b, d are tested below, while c can be added on additionally in the future
+   */
+  // Equivalence class tested: field "name" is string > 100 characters (invalid)
+  it("Should return 400 error when name field is a string of more than 100 characters", async () => {
+    req.fields.name =
+      "This name is so so so long that it should be definitely without a doubt more than one hundred characters";
+    await createProductController(req, res);
+
+    // Assertions
+    expect(res.status).toBeCalledWith(400);
+    expect(res.send.mock.lastCall[0].success).toBe(false);
+    expect(res.send.mock.lastCall[0].message).toBe(
+      "Name of product can only be up to 100 characters long"
+    );
+  });
+
+  // Equivalence class tested: field "name" is non-empty string but product name already exists (invalid)
+  it("Should return 400 error when name field corresponds to a product name that already exists", async () => {
+    const mockSameNameProduct = {
+      photo: {
+        data: "somePhotoData",
+        contentType: "image/jpeg",
+      },
+      _id: new ObjectId("67c950e3a73fac8fc46a070a"),
+      name: "Some valid name",
+      slug: "some-valid-name",
+      description: "some desc",
+      price: 1000,
+      category: new ObjectId("679f463b6d15f42289be8cdd"),
+      quantity: 11,
+      shipping: false,
+      createdAt: "2025-03-06T07:38:11.874Z",
+      updatedAt: "2025-03-06T07:38:11.874Z",
+      __v: 0,
+    };
+
+    productModel.findOne = jest.fn().mockResolvedValue(mockSameNameProduct);
+    await createProductController(req, res);
+
+    // Assertions
+    expect(res.status).toBeCalledWith(400);
+    expect(res.send.mock.lastCall[0].success).toBe(false);
+    expect(res.send.mock.lastCall[0].message).toBe(
+      "Product with this name already exists"
+    );
+  });
+
+  // Equivalence class tested: field "name" is non-empty string but product name already exists (invalid)
```

Comment:
Any reason why this test is duplicated?

Reply:
Oh my bad, I think I duplicated this test by accident

Reply:
LGTM after changes!